File: src/main/java/net/legitimoose/bot/LegitimooseBot.java
package net.legitimoose.bot;

import net.fabricmc.api.ModInitializer;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

import java.io.File;
import java.io.IOException;
import java.io.InputStream;

public class LegitimooseBot implements ModInitializer {
    public static final Logger LOGGER = LoggerFactory.getLogger("legitimoose-bot");
    public static Config CONFIG;

    @Override
    public void onInitialize() {
        InputStream defaultConfigFile = LegitimooseBot.class.getResourceAsStream("/assets/legitimoose-bot/config/legitimoosebot.yml");
        try {
            CONFIG = Config.create(new File("config/legitimoosebot.json"), defaultConfigFile);
        } catch (IOException e) {
            LOGGER.error("failed to initialize config", e);
        }
    }
}

File: src/main/kotlin/net/legitimoose/bot/LegitimooseBot.kt
package net.legitimoose.bot

import net.legitimoose.bot.config.SimpleConfig
import org.slf4j.Logger
import org.slf4j.LoggerFactory

object LegitimooseBot {
    private const val MOD_ID = "legitimoose-bot"
    val logger: Logger = LoggerFactory.getLogger(MOD_ID)
    val config: SimpleConfig = SimpleConfig.of("legitimoosebot-config").request()

    fun init() {}
}
